File: iosApp/iosApp/Presentation/Review/ReviewEditAction.swift
//
//  ReviewEditAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/12.
//

import Foundation

enum ReviewEditAction {
    case clickBack
    case selectRating(Int)
    case clickAddPhoto
    case removePhoto(String)
    case changeReviewText(String)
    case selectAtmosphereAnswer(Bool)
    case clickSubmit
    case dismissInfoMessage
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/review/ReviewEditAction.kt
package com.hhp227.concafe.presentation.review

sealed interface ReviewEditAction {
    data object ClickBack : ReviewEditAction
    data class SelectRating(val rating: Int) : ReviewEditAction
    data object ClickAddPhoto : ReviewEditAction
    data class SelectPhoto(val imageUrl: String) : ReviewEditAction
    data object RemovePhoto : ReviewEditAction
    data class ChangeReviewText(val value: String) : ReviewEditAction
    data class ToggleCastTag(val castId: String) : ReviewEditAction
    data class SelectAtmosphereAnswer(val isPositive: Boolean) : ReviewEditAction
    data object ClickSubmit : ReviewEditAction
    data object DismissInfoMessage : ReviewEditAction
}
